core/test/matrix/permutation.cpp

```diff
@@ -0,0 +1,168 @@
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/core/matrix/permutation.hpp>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/range.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+
+
+#include "core/test/utils/assertions.hpp"
+
+
+namespace {
+
+
+class Permutation : public ::testing::Test {
+protected:
+    using i_type = int;
+    using v_type = double;
+    using Vec = gko::matrix::Dense<v_type>;
+    using Csr = gko::matrix::Csr<v_type, i_type>;
+    Permutation()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx(gko::matrix::Permutation<i_type>::create(
+              exec, gko::dim<2>{4, 3}, gko::Array<i_type>{exec, {1, 0, 2, 3}}))
+    {}
+
+
+    static void assert_equal_to_original_mtx(
+        gko::matrix::Permutation<i_type> *m)
+    {
+        auto perm = m->get_permutation();
+        ASSERT_EQ(m->get_size(), gko::dim<2>(4, 3));
+        ASSERT_EQ(m->get_permutation_size(), 4);
+        ASSERT_EQ(perm[0], 1);
+        ASSERT_EQ(perm[1], 0);
+        ASSERT_EQ(perm[2], 2);
+        ASSERT_EQ(perm[3], 3);
+    }
+
+    static void assert_empty(gko::matrix::Permutation<i_type> *m)
+    {
+        ASSERT_EQ(m->get_size(), gko::dim<2>(0, 0));
+        ASSERT_EQ(m->get_permutation_size(), 0);
+    }
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::unique_ptr<gko::matrix::Permutation<i_type>> mtx;
+};  // namespace
```

Comment:
need two blank lines

Reply:
Sorry, I was wrong before.
You do not need these two blank lines.
`}` is class end not namespace end. 
Thus, only remove `// namespace`